packages/charts/src/components/grid.utils.ts

```diff
@@ -104,6 +104,7 @@ export const generateGridLines = ({
         x2: x,
         y2,
         stroke: isZero ? 2 : undefined,
+        'data-testid': `grid-x-${lines.length}`,
```

Comment:
Should be generated in component or change approach to create grid line `id` instead of data attribute.